vm/src/builtins/bytes.rs

```diff
@@ -606,6 +607,17 @@ impl AsSequence for PyBytes {
     };
 }
 
+impl AsNumber for PyBytes {
+    const AS_NUMBER: PyNumberMethods = PyNumberMethods {
+        remainder: Some(|number, other, vm| {
+            Self::number_downcast(number)
+                .mod_(other.to_pyobject(vm), vm)
```

Comment:
```suggestion
                .mod_(other.to_owned(), vm)
```
we use to_pyobject to create PyObjectRef from a concrete type.
Because we already have `PyObject`, `to_owned()` is enough.